molgenis-data-migrate/src/main/java/org/molgenis/data/version/v1_8/Step11ConvertNames.java

```diff
@@ -0,0 +1,885 @@
+	private DataSource dataSource;
+	private static final Logger LOG = LoggerFactory.getLogger(Step11ConvertNames.class);
+
+	public static final String OLD_DEFAULT_PACKAGE = "default";
+	public static final String NEW_DEFAULT_PACKAGE = "base";
+
+	private HashMap<String, String> packageNameChanges = new HashMap<>();
+	private HashMap<String, String> entityNameChanges = new HashMap<>();
+	private HashMap<String, HashMap<String, String>> attributeNameChanges = new HashMap<>();
+	private HashMap<String, HashMap<String, String>> mrefNameChanges = new HashMap<>();
+	private HashMap<String, String> mrefNoChanges = new HashMap<>();
+	private HashMap<String, List<String>> entitiesAttributesIds = new HashMap<>();
+	private HashSet<String> mysqlEntities = new HashSet<>();
+
+	public Step11ConvertNames(DataSource dataSource)
+	{
+		super(10, 11);
+
+		SingleConnectionDataSource singleConnectionDS = null;
+		try
+		{
+			singleConnectionDS = new SingleConnectionDataSource(dataSource.getConnection(), true);
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+		}
+
+		// we use a single connection datasource because InnoDB only allows disabling key-checks for single sessions
+		this.template = new NamedParameterJdbcTemplate(singleConnectionDS);
+		this.dataSource = singleConnectionDS;
+	}
+
+	@Override
+	public void upgrade()
+	{
+		LOG.info("Validating JPA entities...");
+		checkJPAentities();
+
+		setForeignKeyConstraintCheck(false);
+
+		LOG.info("Validating package names...");
+		checkAndUpdatePackages(null, null);
+
+		LOG.info("Validating entity names...");
+		checkAndUpdateEntities();
+
+		LOG.info("Validating attribute names...");
+		checkAndUpdateAttributes();
+
+		LOG.info("Updating tags...");
+		updateTags();
+
+		LOG.info("Updating MySQL tables...");
+		updateEntityTables();
+
+		setForeignKeyConstraintCheck(true);
+
+		try
+		{
+			// manually close the single datasource connection
+			dataSource.getConnection().close();
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+		}
+	}
+
+	/** Updates the names of columns in (mref)tables and updates the table names themselves. */
+	public void updateEntityTables()
+	{
+		// get a list of all the tables
+		HashSet<String> tableNames = Sets.newHashSet();
+		List<Map<String, Object>> tables = template.getJdbcOperations().queryForList("SHOW TABLES");
+
+		// The name of this column (Tables-in-omx) depends on the database name, workaround:
+		tables.forEach(row -> tableNames.add(row.get(row.keySet().iterator().next()).toString()));
+
+		// get the fullNames of all the entities that have changes (in entity or attribute names)
+		HashSet<String> entitiesToChange = Sets.newHashSet();
+		entitiesToChange.addAll(entityNameChanges.keySet());
+		entitiesToChange.addAll(attributeNameChanges.keySet());
+
+		// update all mref tables that have a change in the mref name
+		for (Map.Entry<String, HashMap<String, String>> mrefTable : mrefNameChanges.entrySet())
```

Comment:
Use Map.Entry<String, Map<String, String>> instead of Map.Entry<String, HashMap<String, String>>